full-stack-intro/client/src/components/Todos.tsx

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+/* eslint-disable @typescript-eslint/no-empty-function -- Remove me */
+import { useEffect, useState } from 'react';
+import { PageTitle } from './PageTitle';
+import { TodoList } from './TodoList';
+import { TodoForm } from './TodoForm';
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export function Todos() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function loadTodos() {
+      try {
+        const response = await fetch('/api/todos');
+        if (!response.ok)
+          throw new Error(`Fetch error with status: ${response.status}`);
+        const result = await response.json();
+        setTodos(result);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    loadTodos();
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo: UnsavedTodo) {
+    console.log('newTodo', newTodo);
+    try {
+      const response = await fetch('/api/todos', {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/JSON',
+        },
+        body: JSON.stringify(newTodo),
+      });
+      if (!response.ok)
+        throw new Error(`Fetch error with status: ${response.status}`);
+      const addedTodos = await response.json();
+      setTodos((todos) => [...todos, addedTodos]);
+    } catch (error) {
+      setError(error);
+    }
+  }
+
+  /* Implement toggleCompleted to toggle the completed state of a todo. Hints are at the bottom of the file. */
+  async function toggleCompleted(todo: Todo) {
+    try {
+      const { todoId } = todo;
+      const response = await fetch(`/api/todos/${todoId}`, {
+        method: 'put',
+        headers: {
+          'Content-Type': 'application/JSON',
+        },
+        body: JSON.stringify({
+          task: todo.task,
+          isCompleted: !todo.isCompleted,
+        }),
+      });
+      if (!response.ok)
+        throw new Error(`Fetch error with status: ${response.status}`);
+      const modifiedTodos = await response.json();
```

Comment:
Since it's just one Todo that is returned, it would be best to use a variable name that is singular (like `modifiedTodo`).